tests/integration/upsertGraph.js

```diff
@@ -154,8 +154,8 @@ module.exports = (session) => {
               ]);
             }
 
-            expect(result.$beforeUpdateCalled).to.equal(undefined);
-            expect(result.$afterUpdateCalled).to.equal(undefined);
```

Comment:
Here's an example of an useless update query. The test was here on purpose to make sure useless updates are not made.

Reply:
Yes, but in this case I believe they should be called since the `model1Relation1` was passed.

I can create a second test that verifies that the aren't called if `model1Relation1` is not passed?

Reply:
The model should only be updated if there is something to actually update. If there is `model1Relation1` it doesn't mean that it's parent should necessarily get updated. The parent should get updated if there is a `BelongsToOneRelation` that gets related, inserted or deleted which causes the foreign key in the parent to get update, which requires the parent to be updated 😅 If a model has some other relation or a `BelongsToOneRelation` that only gets updated, the parent shouldn't get updated.